src/ray/gcs/actor_state_accessor_test.cc

```diff
@@ -0,0 +1,166 @@
+      io_service_.run();
+    }));
+  }
+
+  virtual void TearDown() {
+    gcs_client_->Disconnect();
+
+    io_service_.stop();
+    work_thread->join();
+    work_thread.reset();
+
+    gcs_client_.reset();
+
+    ClearTestData();
+  }
+
+ protected:
+  void GenTestData() { GenActorData(); }
+
+  void GenActorData() {
+    for (size_t i = 0; i < 2; ++i) {
+      std::shared_ptr<ActorTableData> actor = std::make_shared<ActorTableData>();
+      ActorID actor_id = ActorID::FromRandom();
+      actor->set_actor_id(actor_id.Binary());
+      JobID job_id = JobID::FromRandom();
+      actor->set_job_id(job_id.Binary());
+      actor->set_state(ActorTableData::ALIVE);
+      actor_datas_[actor_id] = actor;
+    }
+  }
+
+  void ClearTestData() { actor_datas_.clear(); }
+
+  void WaitPendingDone(std::chrono::milliseconds timeout) {
+    WaitPendingDone(pending_count_, timeout);
+  }
+
+  void WaitPendingDone(std::atomic<int> &pending_count,
+                       std::chrono::milliseconds timeout) {
+    while (pending_count != 0 && timeout.count() > 0) {
+      std::chrono::milliseconds interval(10);
+      std::this_thread::sleep_for(interval);
+      timeout -= interval;
+    }
+    EXPECT_EQ(pending_count, 0);
+  }
+
+ protected:
+  ClientOption option_;
+  ClientInfo info_;
+  std::unique_ptr<RedisGcsClient> gcs_client_;
+
+  boost::asio::io_service io_service_;
+  std::unique_ptr<std::thread> work_thread;
+
+  std::unordered_map<ActorID, std::shared_ptr<ActorTableData>> actor_datas_;
+
+  std::atomic<int> pending_count_{0};
+};
+
+TEST_F(ActorStateAccessorTest, AddAndGet) {
+  ActorStateAccessor &actor_accessor = gcs_client_->Actors();
+  size_t log_length = 0;
+  // add
+  for (const auto &elem : actor_datas_) {
+    const auto &actor = elem.second;
+    JobID job_id = JobID::FromBinary(actor->job_id());
+    ++pending_count_;
+    actor_accessor.AsyncAdd(job_id, elem.first, actor, log_length, [this](Status status) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+    });
+  }
+
+  std::chrono::milliseconds timeout(10000);
+  WaitPendingDone(timeout);
+
+  // get
+  for (const auto &elem : actor_datas_) {
+    const auto &actor = elem.second;
+    JobID job_id = JobID::FromBinary(actor->job_id());
+    ++pending_count_;
+    actor_accessor.AsyncGet(job_id, elem.first,
+                            [this](Status status, std::vector<ActorTableData> datas) {
+                              ASSERT_EQ(datas.size(), 1U);
+                              ActorID actor_id = ActorID::FromBinary(datas[0].actor_id());
+                              auto it = actor_datas_.find(actor_id);
+                              ASSERT_TRUE(it != actor_datas_.end());
+                              --pending_count_;
+                            });
+  }
+
+  WaitPendingDone(timeout);
+}
+
+TEST_F(ActorStateAccessorTest, Subscribe) {
+  ActorStateAccessor &actor_accessor = gcs_client_->Actors();
+  std::chrono::milliseconds timeout(10000);
+  // sub
```

Comment:
nit: sub -> subscribe?

Reply:
sub means subscribe

Reply:
> nit: sub -> subscribe?

yes

Reply:
Let's spell it out to avoid confusion :). `sub` has a lot of other meanings as well.